### PR TITLE
Corrects edge case in zero filling modules folder names

### PR DESCRIPTION
**Pull request by joaomcteixeira**
You are about to submit a new Pull Request. Before continuing make sure you read the [contributing guidelines](CONTRIBUTING.md) and you comply with the following criteria:

- [x] You have stick to Python. Talk with us before adding other programming languages to HADDOCK3
- [ ] Your PR is about CNS
- [x] Your code is well documented: proper docstrings and explanatory comments for those tricky parts
- [x] You structured the code into small functions as much as possible. You can use classes if there's a (state) purpose
- [x] code follows our coding style
- [x] You wrote tests for the new code
- [x] `tox` tests pass. *Run `tox` command inside the repository folder*
- [x] `-test.cfg` examples execute without errors. *Inside `examples/` run `python run_tests.py -b`*
- [x] PR does not add any *install dependencies* unless permission granted by the HADDOCK team
- [x] PR does not break licensing
- [ ] Your PR is about writing documentation for already existing code :fire:
- [ ] Your PR is about writing tests for already existing code :godmode:

---

Corrects edge cases in #378 when having 10 modules because module count starts at `0`.

### CODE REVIEW

**Comment by joaomcteixeira** [comment]
all tests pass... accept it @douweschulte if you don't find any error.
but now I am concerned with https://github.com/haddocking/haddock3/issues/382#issuecomment-1090874872

File: src/haddock/libs/libutil.py [hunk 1]
```diff
@@ -293,6 +293,21 @@ def get_number_of_digits(num):
...
+
+    If there are 5 modules, zerofill digits is 1.
+    If there are 10 modules, zerofill digits is 1 because counting
+    starts at 0 (for topoaa).
+    If there are 100 modules, zerofill digits is 2 because counting
+    starts at 0 (for topoaa).
+
+    This function is used in combination with `zero_fill`.
+    """
+    return max(1, get_number_of_digits(len(modules) - 1))
```
Comment (rvhonorato) [inline src/haddock/libs/libutil.py:308]: Why not simply define everything with 3 digits? its already not realistic that 100 modules will be executed anyway

Comment (rvhonorato) [reply src/haddock/libs/libutil.py:308]: would also make it easier to continue-restart. remember that the main guideline is sense and simplicity :P

Comment (joaomcteixeira) [reply src/haddock/libs/libutil.py:308]: Yeah. That was the original idea. But @douweschulte yesterday rose a point on that, and he was right. There are drawbacks to having alternating leading zeros. But yesterday night I thought that it might not be that problem even for #383 after all. Anyway, now we need to accept this PR to finalize and correct #378. 

If in the future we decide to remove the alternating leading zeros, it is just a matter of changing a function name, nothing more.

Comment (douweschulte) [inline src/haddock/libs/libutil.py:308]: Nit-picking here, is the `max(1, ..)` needed? Because the `get_number_of_digits` already returns at least one.

Comment (douweschulte) [reply src/haddock/libs/libutil.py:308]: I would say it is nice to make sure that it can run in any at all times, even with stupidly high numbers of module, because this would be a very stupid reason to crash the execution of Haddock if everything else is just working. In this implementation if there are 100 modules it starts naming at `000` , `001`, .. . But even if you start a run with 2 modules and so get `0` and `1` it would still be very possible to just continue with `002` if you continue-restart with a run with many more modules.

Comment (joaomcteixeira) [reply src/haddock/libs/libutil.py:308]: Yes. But some of these `000` numbers are hardcoded in the modules run files, for which if the name of the folders change, we would need to change also those paths. With `sed` most likely.

Comment (joaomcteixeira) [reply src/haddock/libs/libutil.py]: But, I realized we need to address this path name changing for some examples, even if we use always two leading zeros. Therefore we can safely use the alternating zerofill; it should not hinder the other implementations.

File: src/haddock/libs/libutil.py [hunk 2]
```diff
@@ -293,6 +293,21 @@ def get_number_of_digits(num):
...
+
+    If there are 5 modules, zerofill digits is 1.
+    If there are 10 modules, zerofill digits is 1 because counting
+    starts at 0 (for topoaa).
+    If there are 100 modules, zerofill digits is 2 because counting
+    starts at 0 (for topoaa).
+
+    This function is used in combination with `zero_fill`.
+    """
+    return get_number_of_digits(len(modules) - 1)
```
Comment (joaomcteixeira) [inline src/haddock/libs/libutil.py:308]: Note: the crucial part in this implementation is the `-1` here.

**Review by douweschulte — APPROVED** [review]
